src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java

```diff
@@ -59,43 +60,41 @@ public CommandResult execute(Model model) throws CommandException {
         List<Person> lastShownList = model.getFilteredPersonList();
 
         if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEXES);
         }
 
-        if (!appointment.startTime().isBefore(appointment.endTime())) { // start time >= end time
+        if (!startTime.isBefore(endTime)) { // start time >= end time
             throw new CommandException(MESSAGE_INVALID_APPOINTMENT_TIME);
         }
 
-        if (appointment.date().isBefore(LocalDate.now()) // date < current date
-                || appointment.date().equals(LocalDate.now()) // date == current date
-                && !appointment.startTime().isAfter(LocalTime.now())) { // start time <= current time
+        if (date.isBefore(LocalDate.now()) // date < current date
+                || date.equals(LocalDate.now()) // date == current date
```

Comment:
In Java, `&&` takes precedence over `||`. I think it would be better in this case to use parenthesis to wrap the second condition instead of wrapping the first condition.